proxy/ptunnel/commands/agent_test.go

```diff
@@ -25,6 +34,14 @@ func TestRunAgentCommand(t *testing.T) {
 }
 
 func TestRunAgentCommand_Error(t *testing.T) {
+	tmpDir, err := ioutil.TempDir("", "en")
```

Comment:
Enough instances of this to be a testutil func somewhere?

```
// go_library marked with `testonly = True`
package kcertmock

func ConfigDirForTestFunc(t *testing.T) func(string, ...string) (string, error) {
  t.Helper()
  tmpDir, err := ioutil.TempDir("", "en")
  assert.NoError(t, err)
  return func (string, ...string) (string, error) {
    return filepath.Join(tmpDir, ".config/enkit"), nil
  }
}
```

test code:

```
import (
  "github.com/enfabrica/enkit/lib/kcert/kcertmock" // or whatever
  "github.com/prashantv/gostub"
)

// ...

stubs := gostub.Stub(&kcerts.GetConfigDir, kcertmock.ConfigDirForTestFunc(t))
defer stubs.Reset()
```

Reply:
...although, replacing a public function in a package you happen to know is a dep seems a bit unorthodox. I'm used to seeing functions that need to be mocked in this way in the tests for the same package only, and the var can be private as a result. In dependent packages - usually one has the opportunity to inject the dependency, via a func var or defining an interface and passing either a "real" impl or "test" impl